aptos-move/aptos-vm/src/data_cache.rs

```diff
@@ -179,21 +97,34 @@ impl<'a, S: StateView> MoveResolverExt for StorageAdapter<'a, S> {
     }
 }
 
-impl<'a, S: StateView> ModuleResolver for StorageAdapter<'a, S> {
-    fn get_module(&self, module_id: &ModuleId) -> Result<Option<Vec<u8>>, Error> {
-        // REVIEW: cache this?
-        let ap = AccessPath::from(module_id);
-        Ok(self.get(ap).map_err(|e| e.finish(Location::Undefined))?)
-    }
-}
-
 impl<'a, S: StateView> ResourceResolver for StorageAdapter<'a, S> {
-    fn get_resource(
+    fn get_resource_with_metadata(
         &self,
         address: &AccountAddress,
         struct_tag: &StructTag,
+        metadata: &[Metadata],
     ) -> Result<Option<Vec<u8>>, Error> {
-        Ok(get_any_resource(self, address, struct_tag)?)
+        Ok(get_any_resource(self, address, struct_tag, metadata)?)
+    }
+}
+
+impl<'a, S: StateView> ModuleResolver for StorageAdapter<'a, S> {
+    fn get_module_metadata(&self, module_id: &ModuleId) -> Vec<Metadata> {
+        let module_bytes = match self.get_module(module_id) {
+            Ok(Some(bytes)) => bytes,
+            _ => return vec![],
+        };
+        let module = match CompiledModule::deserialize(&module_bytes) {
```

Comment:
this may result in inconsistent result compared to load through the VM because of the file format version. this function uses the hard coded VERSION_MAX which is probably >= than the one currently enabled via config https://github.com/aptos-labs/aptos-core/blob/725266cc2096b5681a055883a37d16ff27cd7648/third_party/move/move-vm/runtime/src/config.rs#L10

Reply:
Would that not be the same inconsistency in the current version?

Reply:
> Would that not be the same inconsistency in the current version?

That looks like another bug to me. It's great we're catching it now.

Reply:
I made them consistent